libraries/ESP8266mDNS/ESP8266mDNS.cpp

```diff
@@ -261,6 +261,31 @@ bool MDNSResponder::addServiceTxt(char *name, char *proto, char *key, char *valu
   return false;
 }
 
+void MDNSResponder::clearServiceTxt(const char *name, const char *proto){
+	MDNSService* servicePtr;
+	//Find the service
+  for (servicePtr = _services; servicePtr; servicePtr = servicePtr->_next) {
```

Comment:
declare servicePtr inside the for loop
for (MDNSService * servicePtr = _services; ...)